### ISSUE TITLE
twcgi not unregistering producer correctly (possibly related to distrib).

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7592]
|[<img alt="mesozoic's avatar" src="https://avatars.githubusercontent.com/u/458347?s=50" width="50" height="50">](https://github.com/mesozoic)| @mesozoic reported|
|-|-|
|Trac ID|trac#345|
|Type|defect|
|Created|2003-10-16 13:44:30Z|

<details><summary>Searchable metadata</summary>

```
trac-id__345 345
type__defect defect
reporter__mesozoic mesozoic
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__worksforme worksforme
component__web web
keywords__ 
time__1066311870000000 1066311870000000
changetime__1468844486961361 1468844486961361
version__ 
owner__ 
cc__itamarst cc__mesozoic
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="mesozoic's avatar" src="https://avatars.githubusercontent.com/u/458347?s=50" width="50" height="50">](https://github.com/mesozoic)|@mesozoic commented|
|-|-|

```
#!html
<pre>
I use ResourcePublisher and ResourceSubscription to integrate MovableType with
my twisted-web server. (Not just twisted.web, Moshe's twisted-web for Debian.)
Since I upgraded my packages to the latest 1.1.0alpha, I've been getting the
following error on the ResourceSubscription side:

2003/10/16 03:34 EDT [Broker,client] Peer will receive following PB traceback:
2003/10/16 03:34 EDT [Broker,client] Traceback (most recent call last):
          File "/usr/lib/python2.2/site-packages/twisted/spread/banana.py", line
320, in dataReceived
            processed = cBanana.dataReceived(self.state, buffer,
self.callExpressionReceived)
          File "/usr/lib/python2.2/site-packages/twisted/spread/banana.py", line
93, in callExpressionReceived
            self.expressionReceived(obj)
          File "/usr/lib/python2.2/site-packages/twisted/spread/pb.py", line
563, in expressionReceived
            method(*sexp[1:])
          File "/usr/lib/python2.2/site-packages/twisted/spread/pb.py", line
874, in proto_message
            self._recvMessage(self.localObjectForID, requestID, objectID,
message, answerRequired, netArgs, netKw)
        --- &lt;exception caught here> ---
          File "/usr/lib/python2.2/site-packages/twisted/spread/pb.py", line
888, in _recvMessage
            netResult = object.remoteMessageReceived(self, message, netArgs, netKw)
          File "/usr/lib/python2.2/site-packages/twisted/spread/flavors.py",
line 224, in remoteMessageReceived
            state = apply(method, (self.perspective,)+args, kw)
          File "/usr/lib/python2.2/site-packages/twisted/web/server.py", line
310, in view_registerProducer
            self.registerProducer(_RemoteProducerWrapper(producer), streaming)
          File "/usr/lib/python2.2/site-packages/twisted/protocols/http.py",
line 570, in registerProducer
            self.transport.registerProducer(producer, streaming)
          File "/usr/lib/python2.2/site-packages/twisted/internet/abstract.py",
line 234, in registerProducer
            raise RuntimeError("cannot register two producers")
        exceptions.RuntimeError: cannot register two producers

Any ideas why this is happening? After viewing the site once, the server using
the ResourceSubscription must be restarted.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
It's not a bug, it's a feature.
Your resource publisher has a bug -- it tried to register
two producers without unregistering the first. If you wish
further help, you'll need to post the slave code too, not just
the master.

I'm closing this bug.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
I checked in a possible fix, please try with CVS?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Added release 1.1 tag.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
1. The problem is still there in alpha3
2. Apparently, restarting the server fixes it
3. You don't need PB to create the problem
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
Please run cvs and see if it solves it? I'm hoping i finally got it not broken.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="mesozoic's avatar" src="https://avatars.githubusercontent.com/u/458347?s=50" width="50" height="50">](https://github.com/mesozoic)|@mesozoic commented|
|-|-|

```
#!html
<pre>
I'm using the latest Debian build (1.1.0alpha3) and updated my .tpy's to use the
new Application, and _my_ problem is gone. Whether the real bug is still
occurring in other situations, I don't know.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
I'm not seeing the bug anymore on tm.com
radix should probably peek at the long, and if we don't see
it I think we should pronounce it done-cbb and downgrade to "bug"
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
There's still a bug (the one in the title), but it doesn't affect users anymore.
Downgrading and removing release1.1 topic.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
Why doesn't this affect users any more?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
I did some change to the transport to make this behaviour be bad rather than
fatal... i forget what :/
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun set status to `closed`|
|-|-|

Well, Itamar's a big smelly jerk for not keeping track of what was going on here.

However, three years since the last comment here.  I'm going to go out on a limb and say this isn't a practical issue for anyone anymore.  If someone does run into the problem, they can file a new ticket with more details and we can have another go at fixing it, hopefully a bit more carefully than last time. ;)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_14"></a>|GitHub commented|
|-|-|

In changeset 5caaa11cea8edad2386c6170f73b1830fa8cc5d4
```
#!CommitTicketReference repository="" revision="5caaa11cea8edad2386c6170f73b1830fa8cc5d4"
Merge pull request #345 from twisted/8616-rodrigc-sets

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8616

Do not reference sets module on Python 3
```
